src/shared/treeview/treeNodeUtilities.ts

```diff
@@ -10,29 +10,32 @@ import { AWSTreeNodeBase } from './nodes/awsTreeNodeBase'
  * Produces a list of child nodes using handlers to consistently populate the
  * list when errors occur or if the list would otherwise be empty.
  */
-export async function makeChildrenNodes(parameters: {
-    getChildNodes(): Promise<AWSTreeNodeBase[]>
-    getNoChildrenPlaceholderNode?(): Promise<AWSTreeNodeBase>
-    getErrorNode(error: Error, logID: number): Promise<AWSTreeNodeBase>
-    sort?(a: AWSTreeNodeBase, b: AWSTreeNodeBase): number
```

Comment:
Having this as a method causes the type to be treated as covariant (and thus badly typed!)
All callers that were using `sort` weren't type-safe. One could easily pass in a callback expecting a subtype which the signature would gladly accept, even though it could lead to runtime errors.